Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicNotificationBus.h

```diff
@@ -35,8 +35,27 @@ namespace AzToolsFramework
             // Does not trigger individual OnTemplateRemoved events
             virtual void OnAllTemplatesRemoved() {}
         };
-
         using PrefabPublicNotificationBus = AZ::EBus<PrefabPublicNotifications>;
 
+        //! Used to notify when a prefab has been saved to disk.
+        class PrefabSaveNotifications
+            : public AZ::EBusTraits
+        {
+        public:
+            //////////////////////////////////////////////////////////////////////////
+            // EBusTraits overrides
+            static const AZ::EBusHandlerPolicy HandlerPolicy = AZ::EBusHandlerPolicy::Multiple;
+            static const AZ::EBusAddressPolicy AddressPolicy = AZ::EBusAddressPolicy::ById;
+            using BusIdType = TemplateId;
```

Comment:
Just a reminder: Template ids are not persistent across level loads. So every system listening to this bus should also listen to the PrefabPublicNotifications bus to disconnect from this bus if templates are removed. And in the future listen to RemoveTemplate() notification. Maybe we can call this PrefabTemplateNotifications and reuse it for more things like, creation, reloading, deletion and saving as well?

Reply:
Yeah I was thinking the same thing... but we don't really have another identifier for a prefab template right now (can't even use assetIds since they're source files). Maybe we could use the path, but it sounds very prone to issues...

I can definitely add more notifications if we can make this system more stable, I honestly have a lot of issues with detecting when a new root template is set since the current notifications aren't always triggered consistently.

Reply:
I think using TemplateIds is fine because we don't want systems to stay connected to templates from previous level when they switch levels. My suggestion was to rename this to 'PrefabTemplateNotifications' instead of 'PrefabSaveNotifications' so that we can expand this bus in the future. So, when a template gets removed, the systems can disconnect from the bus accordingly.

On the second point, if there are difficulties tracking level loads, we can discuss introducing a new notification from PrefabEOS. Although, I'd like to discuss more on that outside the scope of this PR. Feel free to reach out to me.

Reply:
Renamed the bus. Will discuss the other issues separately.